### ISSUE TITLE
at-batch: Handle empty iteration spaces gracefully

### ISSUE THREAD

**Issue by ranocha** [JuliaSIMD/CheapThreads.jl#16]
Did I say that [I will probably have quite some loops for `@batch`](https://github.com/JuliaSIMD/CheapThreads.jl/issues/15#issuecomment-813045952) :wink:?
```julia
julia> using Pkg; Pkg.status("CheapThreads")
      Status `~/.julia/environments/v1.6/Project.toml`
  [b630d9fa] CheapThreads v0.1.10

julia> using CheapThreads

julia> Threads.nthreads()
2

julia> function issue16!(dest)
           @batch for i in 2:1
               dest[i] = i
           end
       end
issue16! (generic function with 1 method)

julia> dest = zeros(100);

julia> issue16!(dest); dest
ERROR: 
signal (11): Segmentation fault
in expression starting at none:0
jl_table_peek_bp at /buildworker/worker/package_linux64/build/src/iddict.c:126
jl_eqtable_get at /buildworker/worker/package_linux64/build/src/iddict.c:161
get at ./iddict.jl:87 [inlined]
preserve_handle at ./libuv.jl:57 [inlined]
uv_write at ./stream.jl:983
unsafe_write at ./stream.jl:1064
unsafe_write at ./io.jl:361 [inlined]
write at ./strings/io.jl:185 [inlined]
print at ./strings/io.jl:187 [inlined]
#with_output_color#814 at ./util.jl:91
with_output_color##kw at ./util.jl:71 [inlined]
#printstyled#815 at ./util.jl:105 [inlined]
printstyled##kw at ./util.jl:105 [inlined]
display_error at ./client.jl:102
jfptr_display_error_39872.clone_1 at ~/Software/julia-1.6.0/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1703 [inlined]
jl_f__call_latest at /buildworker/worker/package_linux64/build/src/builtins.c:714
#invokelatest#2 at ./essentials.jl:708 [inlined]
invokelatest at ./essentials.jl:706 [inlined]
print_response at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:242
#40 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:231
jfptr_YY.40_18396.clone_1 at ~/Software/julia-1.6.0/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
with_repl_linfo at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:462
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
print_response at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:229
do_respond at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:798
jfptr_do_respond_19380.clone_1 at ~/Software/julia-1.6.0/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1703 [inlined]
jl_f__call_latest at /buildworker/worker/package_linux64/build/src/builtins.c:714
#invokelatest#2 at ./essentials.jl:708 [inlined]
invokelatest at ./essentials.jl:706 [inlined]
run_interface at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2441
jfptr_run_interface_18797.clone_1 at ~/Software/julia-1.6.0/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
run_frontend at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1126
#44 at ./task.jl:406
jfptr_YY.44_18222.clone_1 at ~/Software/julia-1.6.0/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2237 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2419
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1703 [inlined]
start_task at /buildworker/worker/package_linux64/build/src/task.c:839
unknown function (ip: (nil))
Allocations: 4698723 (Pool: 4697811; Big: 912); GC: 5
Segmentation fault (core dumped)
```

**Comment by ranocha** [comment]
(By the way, the exact error message is highly non-deterministic, but that's probably expected for such a segfault)

**Comment by chriselrod** [comment]
Thanks, I've added it to the tests, but the correct fix is:
https://github.com/JuliaArrays/ArrayInterface.jl/pull/138

**Comment by ranocha** [comment]
Thanks again for this quick fix :+1:

**Comment by ranocha** [comment]
Fixed after updating VectorizationBase v0.19.26 ⇒ v0.19.27
